Fixed #35727 -- Added HttpResponse.text property. (#18534)
django/django: docs/ref/request-response.txt

    A bytestring representing the content, encoded from a string if necessary.

.. attribute:: HttpResponse.text

    .. versionadded:: 5.2

>     A string representation of :attr:`HttpResponse.content`.

Reviewer: Thank you for the quick update! A quick second thought (sorry, you were faster than me) is the following: I think it would be useful to provide details of the encoding used to decode the string.

How about:
```suggestion
    A string representation of :attr:`HttpResponse.content`, decoded using the
    response's :attr:`HttpResponse.charset` (defaulting to UTF-8 if empty).
```

Reviewer 2: Yeah I agree this is nicer :+1:

  :attr:`~StreamingHttpResponse.streaming_content` attribute. This can be used
  in middleware to wrap the response iterable, but should not be consumed.

> * It has no ``text`` attribute. This is because :class:`StreamingHttpResponse` is designed
>   for efficient memory usage by streaming content directly to the client. If you need
>   to access the response content as text, you can use the :attr:`~StreamingHttpResponse.getvalue` method to load
>   the entire content into memory and then convert it to text. However, be cautious as
>   this approach can lead to high memory usage for large content.

Reviewer 2: I think we already describe why the content shouldn't be iterated/consumed 
```suggestion
* It has no ``text`` attribute, as it would require iterating the response object.
```

Reviewer 3: Thank you for summarizing the key points so well.

    This is useful for middleware needing to wrap
    :attr:`StreamingHttpResponse.streaming_content`.

> .. attribute:: StreamingHttpResponse.getvalue()
> 
>     Returns the value of :attr:`StreamingHttpResponse.streaming_content`. This method makes
>     an :class:`StreamingHttpResponse` instance a stream-like object.

Reviewer 2: I'm not sure if we should document this :thinking: 
If we do, maybe...
```suggestion
.. method:: StreamingHttpResponse.getvalue()

    The iterated response content, concatenated into a bytestring encoded
    according to :attr:`HttpResponse.charset`.
```

Reviewer 3: Thank you for letting me know. After giving it some more thought, it doesn't seem appropriate for the current situation. I'll remove that part
